Firestore/core/src/bundle/bundle_serializer.h

```diff
@@ -19,25 +19,81 @@
 
 #include <string>
 #include <utility>
+#include <vector>
 
+#include "Firestore/core/src/bundle/bundle_document.h"
 #include "Firestore/core/src/bundle/bundle_metadata.h"
+#include "Firestore/core/src/bundle/bundled_document_metadata.h"
+#include "Firestore/core/src/bundle/named_query.h"
+#include "Firestore/core/src/core/core_fwd.h"
+#include "Firestore/core/src/model/resource_path.h"
 #include "Firestore/core/src/model/snapshot_version.h"
+#include "Firestore/core/src/remote/serializer.h"
 #include "Firestore/core/src/util/read_context.h"
 #include "nlohmann/json.hpp"
 
 namespace firebase {
 namespace firestore {
 namespace bundle {
 
+class JsonReader : public util::ReadContext {
+ public:
+  const std::string& RequireString(const char* name,
+                                   const nlohmann::json& json_object);
+  const std::vector<nlohmann::json>& RequireArray(
+      const char* name, const nlohmann::json& json_object);
+  const nlohmann::json& Require(const char* child_name,
```

Comment:
Please add a comment for this method. While the name is nice and short, it's not obvious exactly what is required.

Alternatively, you could want to make this `RequireObject` to parallel `RequireArray`, above. This would make it more obvious.